### DISCUSSION TITLE
Swap "Cancel" and "Comment" buttons locations in "Add a comment" dialogue box.

### DISCUSSION THREAD

**Discussion by HaikenEdge** [civitai/civitai#622] [Ideas]
Placing "Cancel" before "Comment in left to right is counter-intuitive, and it's pretty much done in interface design, unless you actually want to discourage commenting and interactions.

**Comment by PeterNjeim** [comment]
While I agree that the `Comment` button should be on the left and the `Cancel` button be on the right, your reasoning for why is wrong. There is no left-to-right intuition when it comes to button placement (or right-to-left intuition for RTL scripts). The only thing that matters is the convention of your operating system. On Windows and some Linux (terminal and Plasma), the action button goes before the cancel button, but on Android, iOS, macOS, and some Linux (GNOME), the action button goes after the cancel button. I said "before" and "after" and not "left" and "right" because RTL scripts show a flipped order for all major operating systems.

I use Windows and obviously prefer the `Comment` button to be before the `Cancel` button since that's what I'm used to, but as long as the action button is made prominent (highlighting it with color), then everything works fine.

Civitai should use the useragent to detect the OS of the user and adjust the order accordingly.
